script.js

```diff
@@ -30,6 +32,37 @@ function setStyles(element) {
 	element.style.left = randInRange(0,leftMax) + 'px';
 	element.style.top = randInRange(0,topMax) + 'px';
 	element.style.backgroundColor = 'rgba(' + randInRange(0,256) + ',' + randInRange(0,256) + ',' + randInRange(0,256) + ',' + Math.random() + ')';
+
+	// using setTimeout, run setStyles on the box once every second, continuously
+	/*var repeater = setTimeout(function(){
+		setStyles(element);
+	},1000);*/
 }
 
-buildBox();
\ No newline at end of file
+var mainDiv = buildBox(),
+	divWidth = mainDiv.offsetWidth,
+	divHeight = mainDiv.offsetHeight,
+	leftMax = windowWidth - divWidth,
+	topMax = windowHeight - divHeight,
+	frameRate = 1000/60,
+	x = randInRange(-5,5),
+	y = randInRange(-5,5); 
+
+// create a new loop that smoothly animates the position of the box, moving it 60 times per second
+function moveBox(element) {
+    var leftPos = element.offsetLeft;
+    	topPos = element.offsetTop;
+
+    if (leftPos > leftMax || leftPos < 0 || topPos > topMax || topPos < 0) {
+        clearTimeout(moveRepeat);
```

Comment:
Don't need this. The setTimeout is made new with each call of moveBox, so there's nothing to cancel.